cmd/docker/builder.go

```diff
@@ -80,12 +86,12 @@ func processBuilder(dockerCli command.Cli, cmd *cobra.Command, args, osargs []st
 		perr = plugin.Err
 	}
 	if perr != nil {
-		// if builder enforced with DOCKER_BUILDKIT=1, cmd fails if plugin missing or broken
-		if enforcedBuilder {
-			return fwargs, fwosargs, newBuilderError(false, perr)
+		// if builder enforced with DOCKER_BUILDKIT=1, cmd must fail if plugin missing or broken
+		if useBuilder {
+			return args, osargs, newBuilderError(false, perr)
 		}
-		// otherwise, display warning and continue
-		_, _ = fmt.Fprintln(dockerCli.Err(), newBuilderError(true, perr))
+		// if not set warning msg as annotation that will be displayed in build cmd
```

Comment:
Micro nit: I liked the `otherwise, ..` here, perhaps:

```suggestion
// otherwise, set the warning msg as annotation that will be displayed in build cmd
```